File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/workflows/cs/createprocessfromtemplate1.cs
// <snippetcreateprocessfromtemplate1>


 // Connect to the Organization service. 
 // The using statement assures that the service proxy will be properly disposed.
 using (_serviceProxy = new OrganizationServiceProxy(serverConfig.OrganizationUri, serverConfig.HomeRealmUri,
                                                      serverConfig.Credentials, serverConfig.DeviceCredentials))
 {
     // This statement is required to enable early-bound type support.
     _serviceProxy.EnableProxyTypes();

     OrganizationServiceContext _orgContext = new OrganizationServiceContext(_serviceProxy);

     CreateRequiredRecords();

     CreateWorkflowFromTemplateRequest request = new CreateWorkflowFromTemplateRequest()
     {
         WorkflowName = "Workflow From Template",
         WorkflowTemplateId = _processTemplateId
     };

     // Execute request.
     CreateWorkflowFromTemplateResponse response = (CreateWorkflowFromTemplateResponse)_serviceProxy.Execute(request);
     _processId = response.Id;

     // Verify success.
     // Retrieve the name of the workflow.
     ColumnSet cols = new ColumnSet("name");
     Workflow newWorkflow = (Workflow)_serviceProxy.Retrieve(Workflow.EntityLogicalName, response.Id, cols);
     if (newWorkflow.Name == "Workflow From Template")
     {
         Console.WriteLine("Created {0}.", request.WorkflowName);
     }

     DeleteRequiredRecords(promptforDelete);
 }

// </snippetcreateprocessfromtemplate1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/workflowsvb/vb/createprocessfromtemplate1.vb
' <snippetcreateprocessfromtemplate1>


 ' Connect to the Organization service. 
 ' The using statement assures that the service proxy will be properly disposed.
 _serviceProxy = ServerConnection.GetOrganizationProxy(serverConfig)
 Using _serviceProxy
     ' This statement is required to enable early-bound type support.
     _serviceProxy.EnableProxyTypes()

     Dim _orgContext As New OrganizationServiceContext(_serviceProxy)

     CreateRequiredRecords()

     Dim request As New CreateWorkflowFromTemplateRequest() With
         {
             .WorkflowName = "Workflow From Template",
             .WorkflowTemplateId = _processTemplateId
         }

     ' Execute request.
     Dim response As CreateWorkflowFromTemplateResponse =
         CType(_serviceProxy.Execute(request), CreateWorkflowFromTemplateResponse)
     _processId = response.Id

     ' Verify success.
     ' Retrieve the name of the workflow.
     Dim cols As New ColumnSet("name")
     Dim newWorkflow As Workflow =
         CType(
             _serviceProxy.Retrieve(
                 Workflow.EntityLogicalName,
                 response.Id,
                 cols), 
             Workflow)
     If newWorkflow.Name = "Workflow From Template" Then
         Console.WriteLine("Created {0}.", request.WorkflowName)
     End If

     DeleteRequiredRecords(promptforDelete)
 End Using

' </snippetcreateprocessfromtemplate1>